flow/envs/green_wave_env.py

```diff
@@ -475,12 +505,13 @@ def observation_space(self):
         """State space that is partially observed.
 
         Velocities, distance to intersections, edge number (for nearby
-        vehicles), and traffic light state.
+        vehicles) from each direction, edge information, and traffic light
+        state.
         """
         tl_box = Box(
             low=0.,
             high=1,
-            shape=(12 * self.num_observed * self.num_traffic_lights +
+            shape=(3 * 4 * self.num_observed * self.num_traffic_lights +
```

Comment:
This is more of a comment for myself for later that we need to name these numbers.

Reply:
Haha, I found myself quite puzzled for a while on this one.